volplugin/utils.go

```diff
@@ -164,3 +189,26 @@ func splitPath(name string) (string, string, error) {
 func joinPath(tenant, name string) string {
 	return strings.Join([]string{tenant, name}, ".")
 }
+
+func addStopChan(name string) chan struct{} {
+	mountMutex.Lock()
+	stopChan := make(chan struct{}, 1)
+
+	if sc, ok := mountStopChans[name]; ok {
+		sc <- struct{}{}
+	}
+
+	mountStopChans[name] = stopChan
+	mountMutex.Unlock()
+
+	return stopChan
+}
+
+func removeStopChan(name string) {
+	if sc, ok := mountStopChans[name]; ok {
+		sc <- struct{}{}
+		mountMutex.Lock()
```

Comment:
Unless I am reading this wrong, I am wondering if we shall take the `Lock` before the if statement itself, otherwise there seems a race where two go routines can still try to delete the same entry and step over each other?